airbyte-webapp-e2e-tests/cypress/pages/replicationPage.ts

```diff
@@ -74,29 +73,90 @@ export const selectSyncMode = (source: string, dest: string) => {
   cy.get(`.react-select__option`).contains(`Source:${source}|Dest:${dest}`).click();
 };
 
-export const selectCursorField = (value: string) => {
-  cy.get(cursorFieldDropdown).first().click({ force: true });
-
-  cy.get(`.react-select__option`).contains(value).click();
+type Dropdown = "cursor" | "primaryKey";
+/**
+ * General function - select dropdown option(s)
+ * @param streamName
+ * @param dropdownType
+ * @param value
+ */
+const selectFieldDropdownOption = (streamName: string, dropdownType: Dropdown, value: string | string[]) => {
+  const container = getFieldDropdownContainer(streamName, dropdownType);
+  const button = getFieldDropdownButton(streamName, dropdownType);
+
+  cy.get(container).within(() => {
+    cy.get(button).click();
+
+    if (Array.isArray(value)) {
+      // in case if multiple options need to be selected
+      value.forEach((v) => cy.get(getFieldDropdownOption(v)).click());
+    } else {
+      // in case if one option need to be selected
+      cy.get(getFieldDropdownOption(value)).click();
+    }
+  });
+  // close dropdown - click on dropdown overlay if exist
```

Comment:
1. In case we select the cursor options - there can be just one option selected, so the dropdown closes automatically. Multiple options can be selected in case we select the PK options, so we need to close the dropdown manually.
Firstly, I did a try to close it when multiple options need to be selected, but since we are in `within()` context, it's strange, but it can find the `body` element. I also tried using `cy.root` and  `cy.document().its('body').find(...)` to find the `Overlay` in the outer scope, but they weren't successful.
2. Hm... But we can have just one Overlay at the same time. The reason why we need to click on Overlay - current implementation dropdown. I also tried to click outside, at any element, and tried on the `body` element - no effect. Only clicking on overlay allows closing the dropdown.
In the case of `Modal` -  we usually have buttons, like "Ok" and "Cancel", and we don't use `Overlay` element. Except for just one case - when we need to check the behavior when we click outside of the `Modal`.

I've added some comments to the code above, why we check Overlay existence.